### ISSUE TITLE
ArrayFieldTemplate - onAddClick should not require an argument

### ISSUE THREAD

**Issue by epicfaace** [rjsf-team/react-jsonschema-form#1314]
### Prerequisites

- [ ] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [ ] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [ ] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://mozilla-services.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

Right now calling this.props.onAddClick() from an ArrayFieldTemplate gives an error because it expects you to pass in an event as an argument. I don't see why passing in an event should be required so we should make it optional.

**Comment by epicfaace** [comment]
Code pointer: https://github.com/mozilla-services/react-jsonschema-form/blob/2a7e4e11fd34952ff4e73cc0e2b39a70356b7dfe/src/components/fields/ArrayField.js#L219

**Comment by bresson** [comment]
I have a fix ready but this appears to be identical to #1324 which appears to be in a holding pattern.

**Comment by epicfaace** [comment]
@bresson sure, feel free to make a PR with the fix + the required tests. You should just be able to make a test with an ArrayFieldTemplate that calls onAddClick with no arguments, then simulate that call, and make sure that no error occurs.

**Comment by bresson** [comment]
@epicfaace Great! As far as I can tell, there are two options: 

Option A
```
  onAddClick = ( event = event ) =>  {
    event.preventDefault();
    ....
  };
```

Option B ( with implied syntheticEvent parameter )
```
  onAddClick () {
    event.preventDefault();
    ....
  };
```
Option A works with all existing tests but is not quite a as elegant as Option B IMHO. However Option B fails about 13 existing tests, mostly referenceErrors for undefined "Event" even when I attempt to pass in the second parameter to Simulate: 

```
  Simulate.click(
    node,
    (event = {
      preventDefault() {}
    })
  );
```
That test snippet works on an isolated test of "react-dom/test-utils" but my guess is unless I'm doing something wrong, perhaps I should spend more time digging through its cousin "react-addons-test-utils" ?

**Comment by bresson** [comment]
@epicfaace Great!

Looks like there are two options : 

Option A
```
  onAddClick = ( event = event ) => {
    event.preventDefault();
    ....
  };
```

Option B ( with implied passing of SyntheticEvent instance )
```
  onAddClick = () => {
    event.preventDefault();
    ....
  };
```
Option A will pass all existing tests. Option B passes manual tests but fails about 13 unit tests, mostly for undefinedReferece error "event." I tried passing in an artificial event object to the unit tests -- 
```
  Simulate.click(
    node,
    (event = {
      preventDefault() {}
    })
  );
``` 
That test snippet should work but I can dig deeper if you want to go with Option B. With Option A, I can update the docs and trigger a PR tomorrow. Thanks!

**Comment by epicfaace** [comment]
I'm not sure what you mean by option b. Where is `event` defined in that case?

**Comment by bresson** [comment]
I'm on shakier ground with Option B but my understanding is onClick will be passed an instance of the SyntheticEvent object. onAddClick would access that param as an implied parameter through lexical scope.

After looking at Option B again, it's less readable and can give the appearance that "event" may be a global. Further, "event" doesn't get the benefit of aliasing through the parameter so in the body it has to be called as "event" ( at least until it is aliased in the body but that's another topic .... ). So sorry this is so long for what should have been a pretty quick PR. Thanks!

**Comment by MParolariDoveconviene** [comment]
I see that this issue has been open for some time. There there are two PRs to fix it #1365 and #1324 but both are "incomplete", missing tests or doc. I created a new PR #1730 that hopefully can be merged. Let me know if something is still missing

**Comment by Parro** [comment]
I see that this issue has been open for some time. There there are two PRs to fix it #1365 and #1324 but both are "incomplete", missing tests or doc. I created a new PR #1730 that hopefully can be merged. Let me know if something is still missing
